pkg/api/customization/monitor/handler_utils.go

```diff
@@ -63,40 +65,102 @@ func (p *clusterGraphInputParser) parseFilter() error {
 		return fmt.Errorf("clusterId is empty")
 	}
 
+	p.ResourceType = p.Input.Filters["resourceType"]
+
+	if p.ResourceType == "" {
+		p.ResourceType = p.Input.Filters["displayResourceType"]
+	}
+
+	if p.ResourceType == "" {
+		return fmt.Errorf("resourceType is empty")
+	}
+
+	clusterResourceTypes := []string{
+		ResourceCluster,
+		ResourceEtcd,
+		ResourceKubeComponent,
+		ResourceRancherComponent,
+		ResourceNode,
+	}
+
+	if !contains(p.ResourceType, clusterResourceTypes...) {
+		return fmt.Errorf("invalid resourceType")
+	}
+
 	for name, value := range p.Input.Filters {
 		p.Conditions = append(p.Conditions, types.NewConditionFromString(name, types.ModifierEQ, value))
 	}
 
 	return nil
 }
 
+func (p *clusterGraphInputParser) parseClusterMetricParams(userContext *config.UserContext, nodeLister v3.NodeLister) error {
+	if p.Input.MetricParams == nil {
+		p.Input.MetricParams = make(map[string]string)
+		return nil
+	}
+
+	var ip string
+	var err error
+	if p.ResourceType == ResourceNode {
+		instance := p.Input.MetricParams["instance"]
+		if instance == "" {
+			return fmt.Errorf("instance in metric params is empty")
+		}
+		ip, err = nodeName2InternalIP(nodeLister, p.ClusterName, instance)
+		if err != nil {
+			return err
+		}
+	}
+	p.Input.MetricParams["instance"] = ip + ".*"
+	return nil
+}
+
+type clusterAuthChecker struct {
+	clusterID  string
+	apiContext *types.APIContext
+}
+
+func newClusterAuthChecker(apiContext *types.APIContext, clusterID string) *clusterAuthChecker {
+	return &clusterAuthChecker{
+		clusterID:  clusterID,
+		apiContext: apiContext,
+	}
+}
+
+func (a *clusterAuthChecker) check() error {
+	var cluster mgmtclient.Cluster
+	return access.ByID(a.apiContext, &managementschema.Version, mgmtclient.ClusterType, a.clusterID, &cluster)
```

Comment:
It is not good to check the privileges by accessing to the cluster object. Maybe we need to use something like https://github.com/rancher/rancher/blob/master/pkg/api/customization/cluster/actionhandler.go#L52-L58 to check the privileges.